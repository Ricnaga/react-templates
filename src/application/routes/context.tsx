import { RouterProvider } from 'react-router-dom';

import { router } from './routes';

export function RouterContext() {
  return <RouterProvider router={router} />;
}
