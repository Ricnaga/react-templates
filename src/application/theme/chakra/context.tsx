import { ChakraProvider } from '@chakra-ui/react';
import { ReactNode } from 'react';

type ChakraUIProps = Partial<Record<'children', ReactNode | JSX.Element>>;

export function ChakraUI({ children }: ChakraUIProps) {
  return <ChakraProvider resetCSS>{children}</ChakraProvider>;
}
