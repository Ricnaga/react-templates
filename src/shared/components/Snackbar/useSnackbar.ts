import { useToast } from '@chakra-ui/react';

type ToastColor = 'info' | 'warning' | 'success' | 'error' | 'loading';

export const useSnackbar = () => {
  const toast = useToast();
  const onSnackBar = (description: string, status: ToastColor = 'success') =>
    toast({
      status,
      get title() {
        return status.toUpperCase();
      },
      description,
      isClosable: true,
      position: 'bottom-left',
    });

  return {
    onSnackBar,
  };
};
