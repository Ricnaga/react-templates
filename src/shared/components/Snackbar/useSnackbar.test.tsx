import { act, renderHook } from '@application/test/testing-library';
import { ThemeProvider } from '@application/theme/chakra/context';

import { useSnackbar } from './useSnackbar';

type IHook = Record<'current', ReturnType<typeof useSnackbar>>;

let resultHook: IHook;

describe('Component: Snackbar', () => {
  beforeEach(() => {
    const { result } = renderHook(() => useSnackbar(), {
      wrapper: ThemeProvider,
    });
    resultHook = result;
  });

  it('should render correctly', () => {
    act(() => resultHook.current.onSnackBar('mock'));
    expect(resultHook).toMatchSnapshot();
  });
});
