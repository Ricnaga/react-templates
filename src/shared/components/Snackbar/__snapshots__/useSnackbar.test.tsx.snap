// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Snackbar should render correctly 1`] = `
{
  "current": {
    "onSnackBar": [Function],
  },
}
`;
